content/en/blog/_posts/2024-03-15-userns-beta/index.md

```diff
@@ -0,0 +1,135 @@
+---
+layout: blog
+title: "Kubernetes 1.30: User-namespace hits Beta!"
+date: 2024-03-15
+slug: userns-beta
+---
+
+**Authors:** Rodrigo Campos Catelin (Microsoft), Giuseppe Scrivano (Red Hat), Sascha Grunert (Red Hat)
+
+Linux provides different namespaces to isolate processes from each other. A
+typical Kubernetes pod runs with a network namespace, to isolate the network
+identity, a PID namespace, to isolate the processes, etc.
+
+One Linux namespace that was left behind is the user namespace. This namespace
+allows to isolate the UIDs and GIDs we use inside the container from the ones on
+the host.
+
+This is a powerful abstraction that allows to run containers as "root": we are
+root inside the container and we can do everything root can inside the pod,
+but our interactions with the host are limited to what a non-privileged user can
+do. This is great to limit the impact of a container breakout.
+
+In Kubernetes v1.25 we introduced support for user namespaces for only stateless
+pods. Kubernetes 1.28 lifted that restriction and now, with Kubernetes 1.30, we
+are moving to beta!
+
+## What is a user namespace?
+
+A user namespace is a Linux feature that isolates the user and group identifiers
+(UIDs and GIDs) of the containers from the ones on the host. The identifiers in
```

Comment:
We use those abbreviations in the first paragraph and think we should introduce them there.

Reply:
Good idea, moved it there.